### PR TITLE
update to ink! 3.4.0

### PR DESCRIPTION
**Pull request by PierreOssun**
- update ink! to 3.4.0
- update openbrush to 2.3.0
- update typechain

### CODE REVIEW

File: uniswap-v2/logics/impls/wnative/mod.rs [hunk 1]
```diff
@@ -15,14 +14,14 @@ impl<T: Storage<psp22::Data> + psp22::Internal> Wnative for T {
...
+        self._mint_to(caller, transfer_value)
     }
 
     default fn withdraw(&mut self, amount: Balance) -> Result<(), PSP22Error> {
         let caller = Self::env().caller();
         self._burn_from(caller, amount)?;
         Self::env()
             .transfer(caller, amount)
-            .map_err(|_| PSP22Error::Custom(String::from("WNATIVE: transfer failed")))
+            .map_err(|_| PSP22Error::Custom("WNATIVE: transfer failed".as_bytes().to_vec()))
```
Comment (HyunggyuJang) [inline uniswap-v2/logics/impls/wnative/mod.rs:25]: Suffices to import OB’s String type.

File: tests/router.ts [hunk 2]
```diff
@@ -1,13 +1,14 @@
...
-import * as BN from "bn.js";
+import BN from "bn.js";
 
 const Decimal = 18
 const ONE = new BN(10).pow(new BN(Decimal))
 
 describe('ROUTER', () => {
     async function setup() {
         const wallet = await getWallet()
+        const wNative = await setupContract('w_native_token', 'new')
```
Comment (HyunggyuJang) [inline tests/router.ts:11]:
```suggestion
        const wNative = await setupContract('wnative_contract', 'new')
```

**Comment by HyunggyuJang** [comment]
I’ll fix typechain tests and push to this branch today.

**Comment by HyunggyuJang** [comment]
Also fixed failing unit tests for psp34 showcase.

**Review by Maar-io — APPROVED** [review]
LGTM

**Pull request merged**